src/resolvers/donationResolver.ts

```diff
@@ -935,6 +939,29 @@ export class DonationResolver {
           ),
         );
       }
+
+      // Validate QF round if provided
+      if (roundId) {
+        const qfRound = await findQfRoundById(roundId);
+        if (!qfRound) {
+          throw new Error('QF round not found');
+        }
+        if (!qfRound.isActive) {
+          throw new Error('QF round is not active');
+        }
+        const now = new Date();
+        if (now < qfRound.beginDate || now > qfRound.endDate) {
+          throw new Error('QF round is not currently active');
+        }
+        // Check if project is in the QF round
+        const projectInQfRound = project.qfRounds?.some(
+          qr => qr.id === roundId,
+        );
+        if (!projectInQfRound) {
+          throw new Error('Project is not part of the specified QF round');
```

Comment:
We should not throw errors because the donation was done already. Then we would need to insert it later. Instead just keep it out of the qfRound by not assigining the qf Round Id, maybe we can add a field for explaining the reason? But ideally dont throw errors here.